content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2422 @@
+detect counter resets based on the whole histogram (see the [counter reset
+considerations section](#counter-reset-considerations) below for details).
+(Note that this always has been a problem for the _sum_ component of classic
+histograms and summaries, too. The approach so far was to accept that counter
+reset detection silently breaks for _sum_ in those cases. Fortunately, negative
+observations are a very rare use case for Prometheus histograms and summaries.)
+
+### Schema
+
+The _schema_ is a signed integer value with a size of 8 bits (short: int8). It
+defines the way bucket boundaries are calculated. The currently valid values
+are -53 and the range between and including -4 and +8. More schemas may be
+added in the future. -53 is a schema for so-called _custom bucket boundaries_
+or short _custom buckets_, while the other schema numbers represent the
+different standard exponential schemas (short: _standard schemas_).
+
+The standard schemas are mergeable with each other and are RECOMMENDED for
+general use cases. Larger schema numbers correspond to higher resolutions.
+Schema _n_ has half the resolution of schema _n_+1, which implies that a
+histogram with schema _n_+1 can be converted into a histogram with schema _n_
+by merging neighboring buckets.
+
+For any standard schema _n_, the boundaries of a bucket with index _i_
+calculated as follows (using Python syntax):
+
+- The upper inclusive limit of a positive bucket:
+  `(2**2**-n)**i`
+- The lower exclusive limit of a positive bucket:
+  `(2**2**-n)**(i-1)`
+- The lower inclusive limit of a negative bucket:
+  `-((2**2**-n)**i)`
+- The upper exclusive limit of a negative bucket:
+  `-((2**2**-n)**(i-1))`
+
+_i_ is an integer number that may be negative.
+
+There are exceptions to the rules above concerning the largest and smallest
+finite values representable as a float64 (called `MaxFloat64` and `MinFloat64`
+in the following) and the positive and negative infinity values (`+Inf` and
+`-Inf`):
+
+- The positive bucket that contains `MaxFloat64` (according to the boundary
+  formulas above) has an upper inclusive limit of `MaxFloat64` (rather than the
+  limit calculated by the formulas above, which would overflow float64).
+- The next positive bucket (index _i_+1 relative to the bucket from the
+  previous item) has a lower exclusive limit of `MaxFloat64` and an upper
+  inclusive limit of `+Inf`. (It could be called a _positive overflow bucket_.)
+- The negative bucket that contains `MinFloat64` (according to the boundary
+  formulas above) has a lower inclusive limit of `MinFloat64` (rather than the
+  limit calculated by the formulas above, which would underflow float64).
+- The next negative bucket (index _i_+1 relative to the bucket from the
+  previous item) has an upper exclusive limit of `MinFloat64` and an lower
+  inclusive limit of `-Inf`. (It could be called a _negative overflow bucket_.)
+- Buckets beyond the `+Inf` and `-Inf` buckets described above MUST NOT be used.
```

Comment:
Using the `positive overflow bucket` and `negative overflow bucket` might be more accurate than `+Inf` and `-Inf` buckets?

Reply:
Those terms are only introduced in parentheses with "they could be called". I'd rather refer to the limits as that's firmly established before. It's also very accurate because the only observed value that will go into the +Inf bucket is, well, +Inf, and likewise for -Inf.